### PR TITLE
add support for gcm apn interface

### PR DESCRIPTION
**Pull request by idoshamun**
add notification property to the notification request body.

https://developers.google.com/cloud-messaging/ios/client

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

File: lib/providers/gcm.js [hunk 1]
```diff
@@ -70,6 +70,7 @@ GcmProvider.prototype._createMessage = function(notification) {
   Object.keys(notification).forEach(function (key) {
     if (notification[key] !== null && typeof notification[key] !== 'undefined') {
       message.addData(key, notification[key]);
+      message.addNotification(key, notification[key]);
```
Comment (raymondfeng) [inline lib/providers/gcm.js]: Do we need somehow distinguish `data` vs `notification`?

Comment (idosh) [inline lib/providers/gcm.js]: The notification field is used for the new gcm apn interface.
I assumed that the notification should be the same for both android and ios.

Comment (raymondfeng) [inline lib/providers/gcm.js]: I was wondering if we should allow the notification argument to have separate data and notification properties, for example:
```js
{
  x: 1,
  notifications: {
    y: 2
  }
}
```

Comment (idosh) [inline lib/providers/gcm.js]: The only thing you should be aware of is that the notification must contain
body and title properties

Comment (raymondfeng) [inline lib/providers/gcm.js]: So do you want to adjust your PR to support the following convention?

```
{
  x: 1,
  notifications: {
    title: 'My msg',
    body: 'This is the full msg'
  }
}
```

Comment (idosh) [inline lib/providers/gcm.js]: I think it's good enough as it is.
No need to separate the notification payload for android and ios.

Comment (superkhau) [inline lib/providers/gcm.js]: @raymondfeng Is this good enough as is?

**Comment by superkhau** [comment]
@idoshamun Can you add some tests to verify your changes and prevent regressions in the future?

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by superkhau** [comment]
@idoshamun 
Closing as abandoned. Please ping me and we can reopen this PR if you want to take it on again.

**Pull request closed without merging**